jina/clients/python/request.py

```diff
@@ -117,14 +117,35 @@ def _generate(data: Union[
                 req.queryset.extend([top_k_queryset])
 
         for content in batch:
-            _add_document(request=req,
-                          content=content,
-                          mode=mode,
-                          docs_in_same_batch=batch_size,
-                          mime_type=mime_type,
-                          buffer_sniff=buffer_sniff,
-                          override_doc_id=override_doc_id
-                          )
+            # TODO:
+            if mode != ClientMode.EVAL:
+                d = getattr(req, str(mode).lower()).docs.add()
+                _fill_document(document=d,
+                               content=content,
+                               docs_in_same_batch=batch_size,
+                               mime_type=mime_type,
+                               buffer_sniff=buffer_sniff,
+                               override_doc_id=override_doc_id
+                               )
+            else:
+                assert len(content) == 2, 'You are passing an Evaluation Request without providing two parts (a ' \
+                                          'document and its groundtruth) '
+                d = getattr(req, str(mode).lower()).docs.add()
+                _fill_document(document=d,
+                               content=content[0],
+                               docs_in_same_batch=batch_size,
+                               mime_type=mime_type,
+                               buffer_sniff=buffer_sniff,
+                               override_doc_id=override_doc_id
+                               )
+                groundtruth = getattr(req, str(mode).lower()).groundtruths.add()
+                _fill_document(document=groundtruth,
+                               content=content[1],
+                               docs_in_same_batch=batch_size,
+                               mime_type=mime_type,
+                               buffer_sniff=buffer_sniff,
+                               override_doc_id=override_doc_id
+                               )
```

Comment:
this looks unnecessary to me so far (reviewing in top-down manner) and problematic:
- it calls `_fill_document` two times, may give the same doc different doc id.
- it should `_fill_document` once and the groundtruth is just a copy of it.

Reply:
note that the groundtruth is filled from different `content` (content[1]) since they come here in Tuples. I do agree it is quite ugly though

Reply:
but there is no point for feeding `content[0] & [1]`. You just feed groundtruth. That's it. The inference procedure won't use them and will override it. This is important.

Reply:
Well,

If I feed a groundtruth with expected `matches` and copy to `docs` I see the problem that every SearchDriver calls `doc.matches.add()` without clearing the document matches. 

So what part of the groundtruth do I need to dismiss? I think it is safer to feed them separately

Reply:
> If I feed a groundtruth with expected matches and copy to docs I see the problem that every SearchDriver calls doc.matches.add() without clearing the document matches.

Then this is the driver problem.

Reply:
but let's keep it in this way. I think this may require quite some refactoring just for enabling this `single-input` feature

Reply:
Well if we let the driver clean the matches then u cannot collect matches from different indexers